### PR TITLE
Fix for overlap hover effect in circular mode view.

### LINKED ISSUE #974: Overlapping of icon-background on hover
web-activity-icons overlap a little with the icons near them on hovering.

sug1     sug2             

sug3

However, it looks fine if the border-radius is set to 50%

ans1

https://user-images.githubusercontent.com/77936051/150928724-56c16ebb-5edc-4cd9-b03b-96535c25aac8.mp4

I would be happy to create a PR if the issue is approved.

### PR DESCRIPTION
**Pull request by AnandChourasia007**
Fixes #974

### CODE REVIEW

**Comment by llaske** [comment]
It doesn't fix the issue.

image

Plus it's not a good idea to limit the number of activities.

**Comment by AnandChourasia007** [comment]
@llaske, please review the latest commit. Instead of hardcoding it now it is being computed using the screen size parameters and the icon spacing and padding parameters. And yes I have checked for the iphone 6,7,8 in the chrome responsive mode too.
Hope this solves the issue.

**Comment by llaske** [comment]
It seems that the spacing is too large now.

image

**Comment by AnandChourasia007** [comment]
@llaske, thanks for the review. please check the latest commit.

File: js/homeview.js [hunk 1]
```diff
@@ -258,7 +258,10 @@ enyo.kind({
 				base_angle = PI2/activitiesCount;
 			} else {
 				restrictedMode = true; spiralMode = false;
-				activitiesCount = parseInt(circumference/icon_padding)-1;
+				activitiesCount = parseInt(circumference/icon_padding);
+				while((circumference/activitiesCount) <= 1.18*constant.iconSpacingFactor*icon_padding){
```
Comment (llaske) [inline js/homeview.js:262]: I suggest to replace the literal `1.18` by a constant.
You could reuse `constant.ringSpaceFactor` in `constant.js`. It's no longer use anywhere.

**Comment by llaske** [comment]
@AnandChourasia007 good job, it's better now.
See my comment above.

**Comment by AnandChourasia007** [comment]
@llaske , thank you for your suggestion. Check the latest commits.

**Comment by llaske** [comment]
@AnandChourasia007 Fine. Thanks for your contribution.

**Pull request merged**